### DISCUSSION TITLE
creating service account for AD user

### DISCUSSION THREAD

**Discussion by tarekabouzeid** [minio/minio#13115] [Q&A]
Hi,

I have a question regarding AD users onboarding to Minio, I have integrated Minio with AD and wanted to create a users onboarding flow. So I want to check if its possible for an AD user to be able to generate a permanent service account credentials "access key & access secret" via HTTP calls or any other method other than logging via web console and do it from there.

What I have tried so far : 

An AD user belongs to an AD group, in Minio we have a default policy attached to this AD group, so this user will always get this policy.

So he will first generate temporary credentials via :
`
curl -X POST "http://localhost:9000?Action=AssumeRoleWithLDAPIdentity&LDAPUsername=myldapuser&LDAPPassword=mypassword&Version=2011-06-15&DurationSeconds=7200"
`

then what is the API call that can generate permanent credentials for him ?  or is there some other way to do that ? 

Best Regards,

**Comment by vadmeste** [comment]
> Then what is the API call that can generate permanent credentials for him ? or is there some other way to do that ?

No, there is no way to have a non expiring credentials. Users information and management are originally controlled by AD, that's why we are only creating temporary accounts here.

**Reply by tarekabouzeid** [reply]
okay thanks for your answer, I have another question please, for example if an AD user want to connect to Minio from Apache spark , where Minio is secured via LDAP. how is that done without generating Minio service account credentials for that AD user?

**Reply by vadmeste** [reply]
> okay thanks for your answer, I have another question please, for example if an AD user want to connect to Minio from Apache spark , where Minio is secured via LDAP. how is that done without generating Minio service account credentials for that AD user?

You still need temporary credentials for this and not a service account.

I did not test it but it should work: https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html#Using_Session_Credentials_with_TemporaryAWSCredentialsProvider

**Reply by tarekabouzeid** [reply]
thanks alot for sharing this,  in case of spark streaming application which runs for long time. the token would expire and application will need to be restarted, thats why i was looking into using persistent access key and access secret which are provided via Minio service accounts that we could create when logging in using the AD user in the web console.

**Reply by vadmeste** [reply]
> , in case of spark streaming application which runs for long time. the token would expire and application will need to be restarted,

This looks like Hadoop/Spark problem. Usually an S3 client will refresh automatically before the expiration time. I suggest that you test TemporaryAWSCredentialsProvider first.
